### PR TITLE
Adding Session Support

### PR DESCRIPTION
**Pull request by marstr**

### CODE REVIEW

**Comment by marstr** [comment]
Derp, I thought strings.Builder was introduced in 1.9. Looks like it wasn't until 1.10. I'll good that next time I'm by a keyboard.

File: message_session.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		return fmt.Errorf("amqp error (%d): %q", rsp.Code, rsp.Description)
+	}
+	return nil
+}
+
+// State retrieves the current State associated with this Session.
+// https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#get-session-state
+func (ms *MessageSession) State(ctx context.Context) ([]byte, error) {
+	link, err := rpc.NewLinkWithSession(ms.receiver.connection, ms.receiver.session.Session, ms.entity.ManagementPath())
+	//link, err := rpc.NewLink(ms.receiver.connection, ms.entity.ManagementPath())
```
Comment (devigned) [inline message_session.go]: delete me

File: message_session_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			defer cancel()
+
+			queueName := suite.randEntityName()
+			cleanup := makeQueue(
+				ctx,
+				t,
+				ns,
+				queueName,
+				QueueEntityWithRequiredSessions(),
+				//QueueEntityWithPartitioning(),
```
Comment (devigned) [inline message_session_test.go]: delete me

File: message_session_test.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			} else {
+				t.Error(err)
+				return
+			}
+
+			const want = "I rode my bicycle past your window last night"
+			msg := NewMessageFromString(want)
+			msg.GroupID = &sessionID
+
+			q.Send(ctx, msg)
```
Comment (devigned) [inline message_session_test.go]:
```suggestion
			suite.NoError(q.Send(ctx, msg))
```

**Pull request merged**